rllib/utils/replay_buffers/tests/test_multi_agent_prioritized_episode_replay_buffer.py

```diff
@@ -0,0 +1,335 @@
+        buffer.add(episode)
+        self.assertTrue(buffer.get_num_episodes() == 1)
+        self.assertTrue(buffer.get_num_timesteps() == 200)
+        self.assertTrue(buffer.get_num_agent_timesteps() == 2 * 200)
+        self.assertTrue(buffer.get_added_agent_timesteps() == 2 * 301)
+        self.assertTrue({eps.id_ for eps in buffer.episodes} == {"E"})
+        for module_id in buffer.get_module_ids():
+            # self.assertTrue(buffer.get_num_episodes(module_id) == 1)
+            self.assertTrue(buffer.get_num_timesteps(module_id) == 200)
+            self.assertTrue(buffer.get_added_timesteps(module_id) == 301)
+
+        # Add another small episode and trigger another eviction.
+        episode = self._get_episode(id_="F", episode_len=2)
+        buffer.add(episode)
+        self.assertTrue(buffer.get_num_episodes() == 1)
+        self.assertTrue(buffer.get_num_timesteps() == 2)
+        self.assertTrue(buffer.get_num_agent_timesteps() == 2 * 2)
+        self.assertTrue(buffer.get_added_agent_timesteps() == 2 * 303)
+        self.assertTrue({eps.id_ for eps in buffer.episodes} == {"F"})
+        for module_id in buffer.get_module_ids():
+            # self.assertTrue(buffer.get_num_episodes(module_id) == 1)
+            self.assertTrue(buffer.get_num_timesteps(module_id) == 2)
+            self.assertTrue(buffer.get_added_timesteps(module_id) == 303)
+
+        # Add N small episodes.
+        for i in range(10):
+            episode = self._get_episode(id_=str(i), episode_len=10)
+            buffer.add(episode)
+        self.assertTrue(buffer.get_num_episodes() == 10)
+        self.assertTrue(buffer.get_num_timesteps() == 100)
+        self.assertTrue(buffer.get_num_agent_timesteps() == 2 * 100)
+        self.assertTrue(buffer.get_added_agent_timesteps() == 2 * 403)
+        for module_id in buffer.get_module_ids():
+            # self.assertTrue(buffer.get_num_episodes(module_id) == 10)
+            self.assertTrue(buffer.get_num_timesteps(module_id) == 100)
+            self.assertTrue(buffer.get_added_timesteps(module_id) == 403)
+
+        # Add a 20-ts episode and expect to have evicted 3 episodes.
+        episode = self._get_episode(id_="G", episode_len=21)
+        buffer.add(episode)
+        self.assertTrue(buffer.get_num_episodes() == 8)
+        self.assertTrue(buffer.get_num_timesteps() == 91)
+        self.assertTrue(buffer.get_num_agent_timesteps() == 2 * 91)
+        self.assertTrue(buffer.get_added_agent_timesteps() == 2 * 424)
+        self.assertTrue(
+            {eps.id_ for eps in buffer.episodes}
+            == {"3", "4", "5", "6", "7", "8", "9", "G"}
+        )
+        for module_id in buffer.get_module_ids():
+            # self.assertTrue(buffer.get_num_episodes(module_id) == 8)
+            self.assertTrue(buffer.get_num_timesteps(module_id) == 91)
+            self.assertTrue(buffer.get_added_timesteps(module_id) == 424)
+
+    # def test_buffer_independent_sample_logic(self):
```

Comment:
Should we include this test (functionality is done in the buffer class, no)?

Reply:
Yes we could include this fucntionality (`get/set_state`).

In a new PR we have to cleanup the sampling tests for "independent" sampling and the `add`  method.